test/core/credentials/transport/tls/spiffe_utils_test.cc

```diff
@@ -34,106 +34,331 @@
-         "contains invalid character B"},
-        {"PathContainsRelativeModifier1", "spiffe://example/path/./foo",
-         ". or .."},
-        {"PathContainsRelativeModifier2", "spiffe://example/path/../foo",
-         ". or .."},
-        {"PathSegmentBadCharacter", "spiffe://example/path/foo.bar/foo@bar",
-         "invalid character @"},
-        {"ContainsNonASCIITrustDomain", "spiffe://µ/path",
-         "cannot contain non-ascii characters"},
-        {"ContainsNonASCIIPath", "spiffe://foo.bar/µ",
-         "cannot contain non-ascii characters"},
-    }),
-    [](const ::testing::TestParamInfo<SpiffeIdFailureTest::ParamType>& info) {
-      return info.param.test_name;
-    });
-
-struct SpiffeIdSuccessTestCase {
-  std::string spiffe_id;
-  std::string trust_domain;
-  std::string path;
-};
-
-using SpiffeIdSuccessTest = TestWithParam<SpiffeIdSuccessTestCase>;
-
-TEST_P(SpiffeIdSuccessTest, SpiffeIdTestSuccess) {
-  const SpiffeIdSuccessTestCase& test_case = GetParam();
+              ::testing::HasSubstr("Path segment contains invalid character"));
+}
+
+TEST(SpiffeId, FragmentInPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo/pa^h").status(),
+            absl::InvalidArgumentError(
+                "Path segment contains invalid character ^. MUST contain only "
+                "letters, numbers, dots, dashes, and underscores"));
+}
+
+TEST(SpiffeId, MultipleSlashesInPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo/bar//baz").status(),
+            absl::InvalidArgumentError("Path segment cannot be empty"));
+}
+
+TEST(SpiffeId, NonASCIIPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo.bar/µ").status(),
+            absl::InvalidArgumentError(
+                "SPIFFE ID URI cannot contain non-ascii characters"));
+}
+
+TEST(SpiffeId, ContainsNonASCIIPathFails) {
+  EXPECT_EQ(SpiffeId::FromString("spiffe://foo.bar/fooµbar").status(),
+            absl::InvalidArgumentError(
+                "SPIFFE ID URI cannot contain non-ascii characters"));
+}
+
+TEST(SpiffeId, NoPathSuccess) {
+  absl::StatusOr<SpiffeId> spiffe_id =
+      SpiffeId::FromString("spiffe://example.com");
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "");
+}
+
+TEST(SpiffeId, BasicSuccess) {
+  absl::StatusOr<SpiffeId> spiffe_id =
+      SpiffeId::FromString("spiffe://example.com/us");
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "/us");
+}
+
+TEST(SpiffeId, WeirdCapitalizationSuccess) {
+  absl::StatusOr<SpiffeId> spiffe_id =
+      SpiffeId::FromString("sPiffe://example.com/us");
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "/us");
+}
+
+TEST(SpiffeId, AllSpiffeCapitalizedSuccess) {
```

Comment:
These next two tests seem like duplicates of the previous test.  Suggest removing.

Reply:
Agreed we can remove this and the next one.